crates/ln-dlc-node/src/ln/manage_spendable_outputs.rs

```diff
@@ -63,19 +68,24 @@ pub fn manage_spendable_outputs<S: TenTenOneStorage, N: Storage>(
         return Ok(());
     }
 
-    let destination_script = wallet.borrow().ldk_wallet().get_last_unused_address()?;
+    let wallet: &OnChainWallet<D> = wallet.borrow();
+    let destination_script = wallet.get_new_address()?;
```

Comment:
In the past we were overusing unused addresses rather than new addresses to avoid having to get a new one from BDK, since the lock was really hard to acquire.

The docs show that it is recommended to use `AddressIndex::LastUnused` judiciously:

```rust
    /// Return the address for the current descriptor index if it has not been used in a received
    /// transaction. Otherwise return a new address as with [`AddressIndex::New`].
    ///
    /// Use with caution, if the wallet has not yet detected an address has been used it could
    /// return an already used address. This function is primarily meant for situations where the
    /// caller is untrusted; for example when deriving donation addresses on-demand for a public
    /// web page.
```

I think this is a spot where we are safe to use a new address: we are building our own transaction to spend a Lightning spendable output.